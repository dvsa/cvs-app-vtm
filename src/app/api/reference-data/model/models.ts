export * from './reference-data.model';
