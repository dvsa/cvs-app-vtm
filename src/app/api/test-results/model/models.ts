export * from './carCancelled';
export * from './carCancelledAdditionalInformation';
export * from './carCancelledAdditionalInformationLocation';
export * from './carCancelledCustomDefects';
export * from './carCancelledDefects';
export * from './carCancelledModType';
export * from './carCancelledTestTypes';
export * from './carCancelledVehicleClass';
export * from './carSubmitted';
export * from './carSubmittedAdditionalInformation';
export * from './carSubmittedDefects';
export * from './carSubmittedTestTypes';
export * from './completeTestResults';
export * from './completeTestResultsVehicleClass';
export * from './customDefect';
export * from './customDefects';
export * from './defect';
export * from './defectAdditionalInformation';
export * from './defectAdditionalInformationLocation';
export * from './defects';
export * from './hgvCancelled';
export * from './hgvSubmitted';
export * from './lgvCancelled';
export * from './lgvSubmitted';
export * from './motorcycleCancelled';
export * from './motorcycleSubmitted';
export * from './motorcycleSubmittedVehicleClass';
export * from './msUserDetails';
export * from './psvCancelled';
export * from './psvCancelledModType';
export * from './psvCancelledTestTypes';
export * from './psvSubmitted';
export * from './psvSubmittedAdditionalInformation';
export * from './psvSubmittedAdditionalInformationLocation';
export * from './psvSubmittedDefects';
export * from './psvSubmittedTestTypes';
export * from './testHistory';
export * from './testResultPutBody';
export * from './testResults';
export * from './testTypeRecords';
export * from './testTypeResults';
export * from './testTypeResultsModType';
export * from './testTypes';
export * from './trlCancelled';
export * from './trlCancelledTestTypes';
export * from './trlSubmitted';
export * from './trlSubmittedTestTypes';
