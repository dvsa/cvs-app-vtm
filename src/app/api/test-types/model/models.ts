export * from './testType';
export * from './testTypeCategory';
export * from './testTypeInfo';
export * from './testTypesTaxonomy';
