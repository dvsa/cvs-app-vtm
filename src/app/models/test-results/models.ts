export * from './completeTestResults';
export * from './completeTestResultsVehicleClass';
export * from './customDefect';
export * from './customDefects';
export * from './defect';
export * from './defectAdditionalInformation';
export * from './defectAdditionalInformationLocation';
export * from './defects';
export * from './testResults';
export * from './testTypeRecords';
export * from './testTypeResults';
export * from './testTypeResultsModType';
export * from './testTypes';
