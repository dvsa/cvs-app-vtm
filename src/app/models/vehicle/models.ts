export * from './adrDetails';
export * from './adrDetailsAdditionalNotes';
export * from './adrDetailsApplicantDetails';
export * from './adrDetailsTank';
export * from './adrDetailsTankTankDetails';
export * from './adrDetailsTankTankDetailsTc2Details';
export * from './adrDetailsTankTankDetailsTc3Details';
export * from './adrDetailsTankTankStatement';
export * from './adrDetailsVehicleDetails';
export * from './applicantDetailsProperties';
export * from './authIntoService';
export * from './axleBrakeProperties';
export * from './axleTyreProperties';
export * from './axleWeightProperties';
export * from './axles';
export * from './brakeForceWheelsNotLocked';
export * from './brakeForceWheelsUpToHalfLocked';
export * from './brakes';
export * from './completeTechRecord';
export * from './completeTechRecordDB';
export * from './completeTechRecordPUT';
export * from './completeTechRecords';
export * from './dda';
export * from './lettersOfAuth';
export * from './manufacturerDetails';
export * from './metadata';
export * from './metadataAdrDetails';
export * from './metadataAdrDetailsAdditionalNotes';
export * from './metadataAdrDetailsTank';
export * from './metadataAdrDetailsTankTankStatement';
export * from './metadataAdrDetailsVehicleDetails';
export * from './microfilm';
export * from './plates';
export * from './platesInner';
export * from './purchaserDetails';
export * from './techRecord';
export * from './techRecordArchiveAndProvisionalPayload';
export * from './techRecordArchiveAndProvisionalPayloadMsUserDetails';
export * from './techRecordBodyType';
export * from './techRecordDimensions';
export * from './techRecordDimensionsAxleSpacing';
export * from './techRecordPOST';
export * from './techRecordPUT';
export * from './techRecordVehicleClass';
export * from './techRecords';
export * from './vrm';
export * from './vrms';
export * from './weights';
