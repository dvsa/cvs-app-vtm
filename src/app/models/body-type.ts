export interface BodyType {
  code: string;
  description: string;
}
