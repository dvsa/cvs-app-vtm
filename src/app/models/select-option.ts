export interface SelectOption {
  id?: number;
  name: string;
  selected: boolean;
}
