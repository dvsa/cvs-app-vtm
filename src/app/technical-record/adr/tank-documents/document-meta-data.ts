export interface DocumentMetaData {
  uuid: string;
  fileName: string;
}
