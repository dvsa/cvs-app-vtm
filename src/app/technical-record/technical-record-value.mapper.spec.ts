import { TestBed } from '@angular/core/testing';

import { BodyType } from '@app/models/body-type';
import { SelectOption } from '@app/models/select-option';
import { SUBSTANCES } from '@app/app.enums';
import { DocumentMetaData } from '../models/document-meta-data';
import { TechnicalRecordValuesMapper } from './technical-record-value.mapper';
import { VehicleTechRecordEdit } from '@app/models/vehicle-tech-record.model';

const axleControlValues = () => {
  return {
    axles: [
      {
        name: 'Axle 1',
        selected: true,
        axleNumber: 1
      }
    ],
    axlesWeights: [
      {
        axleNumber: 1,
        weights: {
          gbWeight: 743,
          eecWeight: 83,
          designWeight: 438
        }
      }
    ],
    axlesTyres: [
      {
        axleNumber: 1,
        tyres: {
          tyreCode: 84,
          tyreSize: 'eu',
          plyRating: '4',
          fitmentCode: 'double',
          dataTrAxles: 8483
        }
      }
    ]
  };
};

const dimensionControlValues = () => {
  return {
    length: null,
    width: null,
    axleSpacing: [
      {
        axles: '1-2',
        from: '1',
        to: '2',
        value: 500
      }
    ]
  };
};

const adrDetailsControlValues = () => {
  return {
    permittedDangerousGoods: [
      {
        name: 'Class 5.1 Hydrogen Peroxide (OX)',
        selected: true
      }
    ] as SelectOption[],
    additionalNotes: {
      number: [
        {
          name: '1A',
          selected: false
        }
      ] as SelectOption[],
      guidanceNotes: true // New Certificate requested
    }
  };
};

const tankDetailsControlValues = () => {
  return {
    tankStatement: {
      substancesPermitted: SUBSTANCES.PERMITTED,
      productListUnNo: ['un1', 'un2', '']
    }
  };
};

const tankDocumentControlValues = () => {
  return [
    {
      metaName: 'UUID_Filename',
      uuid: 'UUID'
    }
  ] as DocumentMetaData[];
};

const vehicleRecordControlValues = () => {
  return {
    techRecord: {
      adrDetails: {
        weight: '5000',
        permittedDangerousGoods: adrDetailsControlValues().permittedDangerousGoods,
        additionalNotes: adrDetailsControlValues().additionalNotes,
        tank: tankDetailsControlValues(),
        memosApply: true, // 07/09 3mth leak ext
        documents: tankDocumentControlValues()
      }
    }
  };
};

describe('TechnicalRecordValuesMapper', () => {
  let vehicleValueMapper: TechnicalRecordValuesMapper;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [TechnicalRecordValuesMapper]
    });

    vehicleValueMapper = TestBed.get(TechnicalRecordValuesMapper);
  });

  describe('mapToAxlesAllowdeValues', () => {
    it('should map AXLE control value in TechRecord to allowed values', () => {
      const controlValues = {
        ...vehicleRecordControlValues(),
        techRecord: {
          ...axleControlValues()
        }
      };
      const result: VehicleTechRecordEdit = vehicleValueMapper.mapControlValuesToDataValues(
        controlValues as any
      );

      expect(result.techRecord[0].axles).toMatchSnapshot();
    });
  });

  describe('mapToDimensionsAllowedValued', () => {
    it('should map DIMENSIONS control value in TechRecord to allowed values', () => {
      const controlValues = {
        ...vehicleRecordControlValues(),
        techRecord: {
          dimensions: dimensionControlValues()
        }
      };
      const result: VehicleTechRecordEdit = vehicleValueMapper.mapControlValuesToDataValues(
        controlValues as any
      );

      expect(result.techRecord[0].dimensions).toMatchSnapshot();
    });
  });

  describe('mapControlValuesToDataValues', () => {
    it('should return null for ADR if not in TechRecord control values', () => {
      const vehicleRecord = {
        techRecord: {
          bodyType: {} as BodyType
        }
      };

      const result: VehicleTechRecordEdit = vehicleValueMapper.mapControlValuesToDataValues(
        vehicleRecord
      );
      expect(result.techRecord[0].adrDetails).toBeNull();
    });

    it('should map ADR control values in TechRecord to allowed values', () => {
      const result: VehicleTechRecordEdit = vehicleValueMapper.mapControlValuesToDataValues(
        vehicleRecordControlValues() as any
      );

      expect(result.techRecord[0].adrDetails).toMatchSnapshot();
    });
  });
});
