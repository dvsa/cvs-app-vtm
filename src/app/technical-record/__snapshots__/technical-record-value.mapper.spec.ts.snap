// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnicalRecordValuesMapper mapControlValuesToDataValues should map ADR control values in TechRecord to allowed values 1`] = `
Object {
  "additionalNotes": Object {
    "guidanceNotes": Array [
      "New certificate requested",
    ],
    "number": Array [],
  },
  "documents": Array [
    "UUID_Filename",
  ],
  "memosApply": Array [
    "07/09 3mth leak ext",
  ],
  "permittedDangerousGoods": Array [
    "Class 5.1 Hydrogen Peroxide (OX)",
  ],
  "tank": Object {
    "tankStatement": Object {
      "productListUnNo": Array [
        "un1",
        "un2",
      ],
      "substancesPermitted": "Substances permitted under the tank code and any special provisions specified in 9 may be carried",
    },
  },
  "weight": "5",
}
`;

exports[`TechnicalRecordValuesMapper mapToAxlesAllowdeValues should map AXLE control value in TechRecord to allowed values 1`] = `
Array [
  Object {
    "axleNumber": 1,
    "parkingBrakeMrk": true,
    "tyres": Object {
      "dataTrAxles": 8483,
      "fitmentCode": "double",
      "plyRating": "4",
      "tyreCode": 84,
      "tyreSize": "eu",
    },
    "weights": Object {
      "designWeight": 438,
      "eecWeight": 83,
      "gbWeight": 743,
    },
  },
]
`;

exports[`TechnicalRecordValuesMapper mapToDimensionsAllowedValued should map DIMENSIONS control value in TechRecord to allowed values 1`] = `
Object {
  "axleSpacing": Array [
    Object {
      "axles": "1-2",
      "value": 500,
    },
  ],
  "length": null,
  "width": null,
}
`;
